client/my-sites/email/google-sale-banner/index.tsx

```diff
@@ -0,0 +1,109 @@
+import canUserPurchaseGSuite from 'calypso/state/selectors/can-user-purchase-gsuite';
+import { getCurrentRoute } from 'calypso/state/selectors/get-current-route';
+import { getSite } from 'calypso/state/sites/selectors';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import type { ResponseDomain } from 'calypso/lib/domains/types';
+
+import './style.scss';
+
+type GoogleSaleBannerProps = {
+	domains: Array< ResponseDomain >;
+};
+
+const GoogleSaleBanner: FunctionComponent< GoogleSaleBannerProps > = ( { domains } ) => {
+	const translate = useTranslate();
+
+	const canCurrentUserPurchaseGSuite = useSelector( canUserPurchaseGSuite );
+	const currencyCode = useSelector( getCurrentUserCurrencyCode );
+	const currentRoute = useSelector( getCurrentRoute );
+	const googleWorkspaceProduct = useSelector( ( state ) =>
+		getProductBySlug( state, GOOGLE_WORKSPACE_BUSINESS_STARTER_YEARLY )
+	);
+
+	const domainsEligibleForGoogleWorkspaceSale = domains
+		.filter( ( domain ) => {
+			if ( domain.expired || domain.isWpcomStagingDomain ) {
+				return false;
+			}
+			if ( ! canCurrentUserAddEmail( domain ) ) {
+				return false;
+			}
+			if ( hasPaidEmailWithUs( domain ) ) {
+				return false;
+			}
+			if ( ! hasGSuiteSupportedDomain( [ domain ] ) ) {
+				return false;
+			}
+
+			return true;
+		} )
+		// Push the primary domain to be listed first
+		.sort( ( a, b ) => Number( b.isPrimary ?? false ) - Number( a.isPrimary ?? false ) );
+
+	const domainForSale = domainsEligibleForGoogleWorkspaceSale[ 0 ] ?? null;
+	const siteForSale = useSelector( ( state ) =>
+		domainForSale?.blogId ? getSite( state, domainForSale.blogId ) : getSelectedSite( state )
+	);
+
+	if ( ! canCurrentUserPurchaseGSuite ) {
+		return null;
+	}
+
+	if ( 0 === domainsEligibleForGoogleWorkspaceSale.length ) {
+		return null;
+	}
+
+	if ( ! hasDiscount( googleWorkspaceProduct ) ) {
+		return null;
+	}
+
+	const monthlyPrice = getMonthlyPrice( googleWorkspaceProduct.sale_cost, currencyCode );
+
+	return (
+		<Banner
+			callToAction={ translate( 'Get Google Workspace' ) }
+			className="google-sale-banner"
+			description={ translate(
+				'Set up a custom email address @%(domainName)s for only {{strong}}%(price)s/mailbox/month{{/strong}} billed annually',
+				{
+					args: {
+						domainName: domainForSale.name,
+						price: monthlyPrice,
+					},
+					comment:
+						'%(domainName)s is a domain name, e.g. example.com; %(price)s is a formatted price, e.g. $3, £2.50',
+					components: {
+						strong: <strong />,
+					},
+				}
+			) }
+			disableCircle
+			iconPath={ googleWorkspaceIcon }
+			href={ emailManagementPurchaseNewEmailAccount(
+				siteForSale?.slug,
+				domainForSale.name,
+				currentRoute,
+				'google-sale'
+			) }
+			title={ translate( 'Get %(discount)d% off Google Workspace for your domain!', {
+				args: { discount: googleWorkspaceProduct?.sale_coupon?.discount ?? 50 },
```

Comment:
It may be better to not show this banner at all if that discount amount is not available.

Reply:
Done. I've added an explicit check for the `sale_coupon.discount` property.